testutil/ciao-down/workload.go

```diff
@@ -75,7 +79,47 @@ func (wkld *workload) save(ws *workspace) error {
 	return nil
 }
 
-func loadWorkloadData(ws *workspace, workloadName string) ([]byte, error) {
+func loadWorkloadData(ctx context.Context, ws *workspace, workloadName string) ([]byte, error) {
+	var absPath string
+	u, err := url.Parse(workloadName)
+	if err == nil {
+		switch u.Scheme {
```

Comment:
This code needs to move into a separate function.  In loadWorkloadData we need  (comments are optional)

```
// Check to see whether workloadName might refer to something in our workload directories
// If this condition fails we know that workloadName cannot be an absolute path.
if filepath.Base(workloadName) != workloadName {
   return workloadFromURL(u, ctx)
}

// Fall through to the existing code.

```

Put all the new URL handling code in workloadFromURL.  It will figure out the correct path, downloading the file if necessary and return either an error or the bytes of the workload.